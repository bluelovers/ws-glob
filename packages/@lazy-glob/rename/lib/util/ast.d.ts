import { IPartAstInterface, IQuestionMark, IAsterisk } from '../types';
export declare function isAsterisk(g: any | IPartAstInterface): g is IAsterisk;
export declare function isQuestionMark(g: any | IPartAstInterface): g is IQuestionMark;
