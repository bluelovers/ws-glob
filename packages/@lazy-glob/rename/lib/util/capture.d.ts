export declare function capture(names: string[], glob: string): import("./globGroupsCollection").globGroupsCollectionInterface[];
