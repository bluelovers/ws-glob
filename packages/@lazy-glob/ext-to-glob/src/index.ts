import { escapeRegExp } from 'regexp-helper-core';


export type IExtString<T extends string = string> = `.${T}`

export function removeDot<T extends string>(ext: IExtString<T>)
{
	return ext.replace(/^\./, '') as T
}

export function removeDotFromExtensions<T extends string>(arr: readonly IExtString<T>[], fns?: ((input: string) => string)[])
{
	return arr.map(input => {
		input = removeDot(input) as any;
		fns?.forEach(fn => {
			input = fn(input) as any;
		})
		return input
	})
}

export function extToGlob<T extends string>(arr: readonly IExtString<T>[])
{
	return `.+(${removeDotFromExtensions(arr).join('|')})` as const
}

export function extToRegexpPattern<T extends string>(arr: readonly IExtString<T>[])
{
	return `.(${removeDotFromExtensions(arr, [escapeRegExp]).join('|')})$` as const
}

export default extToGlob
