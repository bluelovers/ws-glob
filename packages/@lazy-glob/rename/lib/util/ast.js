"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isQuestionMark = exports.isAsterisk = void 0;
const types_1 = require("../types");
function isAsterisk(g) {
    return (g.type === 'wildcard' && g.pattern === types_1.EnumWildcard.Asterisk);
}
exports.isAsterisk = isAsterisk;
function isQuestionMark(g) {
    return (g.type === 'wildcard' && g.pattern === types_1.EnumWildcard.QuestionMark);
}
exports.isQuestionMark = isQuestionMark;
//# sourceMappingURL=ast.js.map