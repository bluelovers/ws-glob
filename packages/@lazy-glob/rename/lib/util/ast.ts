import { IPartAstInterface, IQuestionMark, EnumWildcard, IAsterisk } from '../types';

export function isAsterisk(g: any | IPartAstInterface): g is IAsterisk
{
	return (g.type === 'wildcard' && g.pattern === EnumWildcard.Asterisk);
}

export function isQuestionMark(g: any | IPartAstInterface): g is IQuestionMark
{
	return (g.type === 'wildcard' && g.pattern === EnumWildcard.QuestionMark);
}
