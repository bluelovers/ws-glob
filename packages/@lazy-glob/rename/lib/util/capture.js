"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * For each string in names array, return a globGroupsCollection (an object implementing globGroupsCollectionInterface) containing the substring matched by each part of the glob pattern.
 * Example of parts of a glob: "?omer.*" => Parts: ["?"] ["omer."] ["*"]
 * @method capture
 * @param names {String[]} List of names
 * @param glob {String} The glob pattern to be applied to names
 * @return {globGroupsCollection[] | null} An array of globGroupsCollection objects. Returns null if invalid parameters or names list is empty.
 */
const globGroupsCollection_1 = require("./globGroupsCollection");
function capture(names, glob) {
    if (!Array.isArray(names) || !names.length) {
        return null;
    }
    if ((typeof glob !== 'string') || !glob.length) {
        return null;
    }
    // For each name received, create an object of type 'globGroupsCollection' containing the match for each part of the glob
    return names.map(function buildGlobGroupsCollection(name) {
        let groupsObj = globGroupsCollection_1.globGroupsCollectionFactory();
        groupsObj.initGroups(glob);
        groupsObj.buildGroups(name);
        return groupsObj;
    });
}
//# sourceMappingURL=capture.js.map