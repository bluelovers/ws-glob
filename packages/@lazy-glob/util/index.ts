/**
 * Created by user on 2020/6/9.
 */

export * from './lib/types';

export default exports as typeof import('./index');
