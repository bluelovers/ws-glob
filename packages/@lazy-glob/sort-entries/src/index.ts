/**
 * Created by user on 2020/6/9.
 */

import { naturalCompare } from '@bluelovers/string-natural-compare';
import { ITree } from '@lazy-glob/util/lib/types';
import { ISortEntriesSort, ISortEntriesSortRow } from '@lazy-glob/util/lib/types/entries';
import { SymGlobTree } from '@lazy-glob/util';

type IEntry<T extends Record<any, any>, K extends keyof T> = [K, T[K]]
type IEntries<T extends ITree = ITree> = IEntry<T, keyof T>[];

export type IEntriesInput = IEntries | ISortEntriesSort;

export function entries_sort(entries: IEntriesInput, fn: (a: string, b: string, cache: any) => number = naturalCompare, cache: any = {}): ISortEntriesSort
{
	entries = (entries as ISortEntriesSort).reduce(function (a: ISortEntriesSort, b: ISortEntriesSortRow)
	{
    const v = b[1];

    if (v === null || typeof v == 'string')
		{
			a.push(b);
		}
		else
		{
      const d = Object.entries(v);
      a.push([b[0], entries_sort(d, fn), (v as any)[SymGlobTree]]);
		}

		return a;
	}, [] as ISortEntriesSort);

	(entries as ISortEntriesSort).sort(function (a, b)
	{
    const r = fn(a[0], b[0], cache);
    return r;
	});

	return entries;
}

export function entries_reduce(entries: IEntriesInput)
{
	return (entries as ISortEntriesSort)
		.reduce(function (a, [k, v, bool]: ISortEntriesSortRow)
		{
			if (v === null || typeof v == 'string')
			{
				a[k] = v as string;
			}
			else
			{
				a[k] = entries_reduce(v);
			}

			if (bool)
			{
        // @ts-ignore
				a[k][SymGlobTree] = bool;
			}

			return a;
		}, {} as ITree)
		;
}

export function sort(a: ITree, fn: (a: string, b: string, cache: any) => number = naturalCompare)
{
  const r = Object.entries(a);

  const a1 = entries_sort(r, fn);
  const a2 = entries_reduce(a1);

  return a2
}

export default sort
