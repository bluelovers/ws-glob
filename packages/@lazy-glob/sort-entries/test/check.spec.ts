import { globToTree } from 'glob-tree-list/src/index';
import { inspect } from "util";
import { sort, entries_sort, entries_reduce } from '../src/index';

describe(`no empty`, () =>
{
	[
		[
			'packages/@yarn-tool/ws-changed/lib/',
			'packages/@yarn-tool/ws-changed/',
			'packages/ws-pkg-list/lib/',
			'packages/@yarn-tool/find-root/',
		],
	].forEach(ls =>
	{

		test(inspect(ls), () =>
		{
			let actual = globToTree(ls);

			let r = Object.entries(actual);
			let a1 = entries_sort(r);
			let a2 = entries_reduce(a1);

			let expected = sort(a2);

			expect(actual).toStrictEqual(expected);
			expect(actual).toStrictEqual(a2);

			expect(actual).toMatchSnapshot();
			expect(r).toMatchSnapshot();
			expect(a1).toMatchSnapshot();

		});

	})

})
