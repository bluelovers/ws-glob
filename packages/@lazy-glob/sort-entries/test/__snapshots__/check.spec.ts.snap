// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`no empty [
  'packages/@yarn-tool/ws-changed/lib/',
  'packages/@yarn-tool/ws-changed/',
  'packages/ws-pkg-list/lib/',
  'packages/@yarn-tool/find-root/'
] 1`] = `
{
  "packages/": {
    "@yarn-tool/": {
      "find-root/": {
        Symbol(@lazy-glob/sort-tree): true,
      },
      "ws-changed/": {
        "lib/": {
          Symbol(@lazy-glob/sort-tree): true,
        },
        Symbol(@lazy-glob/sort-tree): true,
      },
    },
    "ws-pkg-list/": {
      "lib/": {
        Symbol(@lazy-glob/sort-tree): true,
      },
    },
  },
}
`;

exports[`no empty [
  'packages/@yarn-tool/ws-changed/lib/',
  'packages/@yarn-tool/ws-changed/',
  'packages/ws-pkg-list/lib/',
  'packages/@yarn-tool/find-root/'
] 2`] = `
[
  [
    "packages/",
    {
      "@yarn-tool/": {
        "find-root/": {
          Symbol(@lazy-glob/sort-tree): true,
        },
        "ws-changed/": {
          "lib/": {
            Symbol(@lazy-glob/sort-tree): true,
          },
          Symbol(@lazy-glob/sort-tree): true,
        },
      },
      "ws-pkg-list/": {
        "lib/": {
          Symbol(@lazy-glob/sort-tree): true,
        },
      },
    },
  ],
]
`;

exports[`no empty [
  'packages/@yarn-tool/ws-changed/lib/',
  'packages/@yarn-tool/ws-changed/',
  'packages/ws-pkg-list/lib/',
  'packages/@yarn-tool/find-root/'
] 3`] = `
[
  [
    "packages/",
    [
      [
        "@yarn-tool/",
        [
          [
            "find-root/",
            [],
            true,
          ],
          [
            "ws-changed/",
            [
              [
                "lib/",
                [],
                true,
              ],
            ],
            true,
          ],
        ],
        undefined,
      ],
      [
        "ws-pkg-list/",
        [
          [
            "lib/",
            [],
            true,
          ],
        ],
        undefined,
      ],
    ],
    undefined,
  ],
]
`;
